src/encoded/static/components/cart/cart.js

```diff
@@ -822,6 +824,66 @@ FileFacets.defaultProps = {
 };
 
 
+/**
+ * Display a button that links to experiment search results. This *only* works for Experiments, not
+ * FCCs for now.
+ */
+const MAX_SEARCHABLE_DATASETS = 125; // Maximum number of datasets for search-results links.
+const CartDatasetSearch = ({ elements }, reactContext) => {
+    const [isWarningVisible, setIsWarningVisible] = React.useState(false);
+
+    // * For now only include experiments. Might need to expand this in the future.
+    const experimentElements = elements.filter(element => element.includes('/experiments/'));
+
+    // Called when the user clicks the Dataset Search button.
+    const handleButtonClick = () => {
+        if (experimentElements.length > MAX_SEARCHABLE_DATASETS) {
+            setIsWarningVisible(true);
+        } else {
+            const datasetsQuery = experimentElements.map(element => `@id=${encodedURIComponent(element)}`).join('&');
+            const query = `/search/?type=Experiment&${datasetsQuery}`;
+            reactContext.navigate(query);
+        }
+    };
+
+    // Called when the user closes the warning modal.
+    const handleCloseClick = () => {
+        setIsWarningVisible(false);
+    };
+
+    // Only display the Dataset Search button if we have at least one experiment in the cart.
+    if (experimentElements.length > 0) {
+        return (
+            <React.Fragment>
+                <button onClick={handleButtonClick} className="btn btn-sm btn-info btn-inline">Dataset search</button>
+                {isWarningVisible ?
+                    <Modal closeModal={handleCloseClick}>
+                        <ModalHeader title={<h4>Cart dataset search results</h4>} closeModal={handleCloseClick} />
+                        <ModalBody>
+                            Viewing cart dataset search results requires {MAX_SEARCHABLE_DATASETS} datasets
+                            or fewer. This cart contains {elements.length} datasets.
```

Comment:
if element.length === 1, it reads- "This cart contains 1 dataset**s**". That is a grammatical error.

Reply:
The only way to have that situation is if the maximum number of allowable elements is zero.